lib/puppet/pops/model/ast.rb

```diff
@@ -0,0 +1,5169 @@
+# # Generated by Puppet::Pops::Types::RubyGenerator from TypeSet Puppet::AST on -4712-01-01
+
+module Puppet
+module Pops
+module Model
+class PopsObject
+  def self._pcore_location
+    @_pcore_location ||= (loc = Puppet::Util.path_to_uri("#{__FILE__}"); URI("#{loc}?line=#{__LINE__.to_i - 3}"))
+  end
+
+  def self._pcore_type
+    @_pcore_type ||= Types::PObjectType.new('Puppet::AST::PopsObject', {
+    })
+  end
+
+  include Types::PuppetObject
+
+  def self.from_hash(i12n)
+    from_asserted_hash(Types::TypeAsserter.assert_instance_of('Puppet::AST::PopsObject initializer', _pcore_type.i12n_type, i12n))
+  end
+
+  def self.from_asserted_hash(i12n)
+    new
+  end
+
+  def self.create
+    new
+  end
+  attr_reader :hash
+
+  def initialize
+    @hash = 925601229154960054
+  end
+  def _pcore_init_hash
+    {}
+  end
+
+  def _pcore_contents
+  end
+
+  def _pcore_all_contents(path)
+  end
+
+  def to_s
+    Types::TypeFormatter.string(self)
+  end
+
+  def eql?(o)
+    o.instance_of?(self.class)
+  end
+  alias == eql?
+end
+
+class Positioned < PopsObject
+  def self._pcore_location
```

Comment:
I'm not so sure that appointing the input source is the right thing to do. It all depends on your use-case. If you're trying to find a problem, then it might be very helpful to see where the actual Ruby class is. After all, the loader doesn't load this type from the .pp file, so stating that is somewhat false.